python/tests/core/ecl/test_pylint.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+#  Copyright (C) 2017 Statoil ASA, Norway.
+#
+#  This file is part of ERT - Ensemble based Reservoir Tool.
+#
+#  ERT is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+#  FITNESS FOR A PARTICULAR PURPOSE.
+#
+#  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+#  for more details.
+
+import fnmatch
+import os
+import unittest
+from pylint import epylint as lint
+from ert.test import ExtendedTestCase
+
+class CodeTestCase(ExtendedTestCase):
+    """This class is a test case for linting."""
+
+    LINT_ARGS = ['-d', 'R,C,W'] + \
+                ['--extension-pkg-whitelist=numpy']
+
+    def _get_lintable_files(self, paths, whitelist=None):
```

Comment:
I would suggest one of two:

1. The two methods `_get_lintable_files()` and `test_linting()` are combined into one method:

```python
def lint_path(self, path, white_list):
     for root, _ . filenames = os.walk( path )
          ...
          # The lint call one single file could maybe be a separate method?
          self.assertEqual( 0 , lint.lint( )) 
```
And then an instance of `CodeTestCase` would call the `lint_path( )` method.

2. Or alternatively the `_get_lintable_files()` method should be `@classmethod` - currently it does not access `self`.